app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -457,7 +467,7 @@ public function handleTransactionUpdate(Mage_Payment_Model_Info $payment, $newTr
             if ($prevTransactionStatus != null) {
                 $prevTransactionStatus = strtolower($prevTransactionStatus);
 
-                if ($newTransactionStatus == $prevTransactionStatus) {
+                if ($newTransactionStatus != self::TRANSACTION_REFUND && $newTransactionStatus == $prevTransactionStatus) {
```

Comment:
1. Bolt does support multiple partial refunds now
2. then for consecutive refunds the prevstatus can be same as the newstatus, that's why I add this exceptional case

Reply:
@dumega In the case of consecutive refunds `$prevTransactionStatus` would be `self::TRANSACTION_REFUND`.  The sentinel value would never be set as it is the `$prevTransactionStatus` and not the `$newTransactionStatus`.  The sentinel value, which is never set, is always guaranteed to be different than the new status.

I think the sentinel way makes the code logic more clear because of uniformity.  This is not a big deal.  People can read the code and figure out what is going on and will just have to ask themselves, "Why is the refund case special?"